pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -356,6 +366,38 @@ func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, boundClaims, claimsToBind []*
 	return
 }
 
+// GetEligibleNodes returns a list of eligible nodes by name based on the bound claims of the pod to
+// potentially reduce down the list of nodes in subsequent scheduling stages.
+func (b *volumeBinder) GetEligibleNodes(boundClaims []*v1.PersistentVolumeClaim) sets.String {
+	if len(boundClaims) == 0 {
+		return sets.NewString()
+	}
+
+	notFound := &errNotFound{}
+	var errs []error
+	eligibleNodes := sets.NewString()
+	for _, pvc := range boundClaims {
+		pvName := pvc.Spec.VolumeName
+		pv, err := b.pvCache.GetPV(pvName)
+		if err != nil && !errors.As(err, &notFound) {
+			errs = append(errs, err)
+			continue
+		}
+
+		// if the PersistentVolume has node affinity matching specific node(s), add them to the eligible nodes
+		nodeNames := util.GetPersistentVolumeNodeNames(pv)
+		for _, node := range nodeNames {
+			eligibleNodes.Insert(node)
```

Comment:
Do we potentially want to take a union of the nodes across all pvs? So if pv1 is node1 and pv2 is node2, then it should return 0 nodes?

Reply:
Building on this, should we distinguish between 0 nodes and all nodes?

Reply:
Thanks @msau42 - I've updated the `GetEligibleNodes()` to:

1. return a bool `canReduceNodeList` along with the eligibleNodes set to indicate whether node reduction has been done, which will only be true if boundClaims bound to at least one local PV. If error occurred during get PV from cache or if no local PV, then this is false and all nodes will be considered

2. Using intersection for the eligibleNodes so for the case of PV1 to node1 and PV2 to node2, then intersection of those will be empty list